src/PrestaShopBundle/Service/MailTemplate/GenerateMailTemplatesService.php

```diff
@@ -0,0 +1,144 @@
+<?php
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Service\MailTemplate;
+
+use PrestaShop\PrestaShop\Core\Exception\InvalidArgumentException;
+use PrestaShop\PrestaShop\Core\Language\LanguageInterface;
+use PrestaShop\PrestaShop\Core\Language\LanguageRepositoryInterface;
+use PrestaShop\PrestaShop\Core\MailTemplate\MailTemplateGenerator;
+use PrestaShop\PrestaShop\Core\MailTemplate\ThemeCatalogInterface;
+use PrestaShop\PrestaShop\Core\MailTemplate\ThemeInterface;
+use PrestaShop\PrestaShop\Core\Exception\FileNotFoundException;
+
+/**
+ * This a useful and easy to use service to generate mail templates. It only has
+ * string parameters and therefore can be easily called via a command, controller
+ * or any other service without much knowledge of the MailTemplate architecture.
+ *
+ * It is also defined as a Symfony service with default parameters so that mails
+ * are exported in the appropriate folders.
+ */
+class GenerateMailTemplatesService
```

Comment:
Would you mind to remove the suffix "Service"? 

I can tell it's a terrible idea regarding what we have done with TranslationService.

If you think this class has one responsibility, name it accordingly and put it in Core namespace.
If this class have multiple responsibilities, we should split it into multiple ones.

Atm I don't see the value of these getters and setters: if the responsibility of this class is to generate mail templates, so be it! This should be the only method available publicly: don't you think?

Also, the `Service` namespace of PrestaShopBundle is deprecated, can I suggest src/Core/MailTemplate/MailGenerator.php instead? :)

Reply:
There is already the `MailTemplateGenerator` class in the core which actually does all the work.
But it expects real object as input `LanguageInterface`, `ThemeInterface`, specify the target folders and so on...

The only purpose of the `GenerateMailTemplatesService` is to provide an easier way to use it with simple string inputs (it is responsible to fetch to objects based on the input strings, it has default target folders set which can be overridden if need be)
This is to simplify the command and the controller that uses it, but I don't think it has its place in the `Core` domain as it would be redundant with `MailTemplateGenerator`